sys/dev/qlnx/qlnxe/qlnx_os.c

```diff
@@ -446,17 +445,12 @@ qlnx_num_tx_compl(qlnx_host_t *ha, struct qlnx_fastpath *fp,
 {
 	u16 hw_bd_cons;
 	u16 ecore_cons_idx;
-	uint16_t diff;
 
 	hw_bd_cons = le16toh(*txq->hw_cons_ptr);
 
 	ecore_cons_idx = ecore_chain_get_cons_idx(&txq->tx_pbl);
-	if (hw_bd_cons < ecore_cons_idx) {
```

Comment:
how do we know this is always false?

Reply:
It isn't always false.
The hardware counter will rollover to zero before the software one but subtracting them will always give the correct value.
e.g. 0 - 0xFFFF = 1 (for unsigned 16 bit integer type)

Reply:
That makes sense.